SalesforceCommerceCloud/pwa-kit
# [retail-react-app] Integrate commerce hooks provider

In order to integrate our commerce hooks for retail-react-app template, we'll need to first integrate the Provider component. 

Because the integration work will be done in stages, to avoid breaking the entire template:
- We can have both the old and the new Commerce API client in there for now. For the old client, this PR renames some things by prepending them with '_'.
- Using both `getProps` and react-query to fetch data. 

[Ticket](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000010oHrhYAE/view)

Related PR #797 

# How to Test-Drive This PR

To verify the JWT token:
1. `npm start` in the template-retail-react-app
1. Navigate to localhost:3000
2. Look into the Local Storage and double check the key `access_token` ← it's set by our commerce hooks
3. On the other hand, the key `_token` is the one set by the old api client.

~~To verify that you can call our commerce hooks, I have added a few calls to these components:~~ Update: I've already removed those hooks.
- Home component: it has a successful product query.
- Header component: it has a failed query.

You can see the above queries in more detail through the React Query DevTools. 

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
